File: ads homeworks solutions/week_8/countSort.py
array = [9, 1, 6, 7, 6, 2, 1]  

def counting_sort(array):
    k = max(array)
    n = len(array)
    output = [0] * n
    temp = [0] * (k + 1)

    for i in range(n):
        temp[array[i]] += 1
    for i in range(1, k + 1):
        temp[i] += temp[i - 1]
    for i in range(n - 1, -1, -1):
        output[temp[array[i]] - 1] = array[i]
        temp[array[i]] -= 1
    return output

print(*counting_sort(array))

File: ads homeworks solutions/week_8/countsort.cpp
#include <iostream>

int main() {
    int array[] = {9, 1, 6, 7, 6, 2, 1};

    int n = 7;
    int output[n]; 
    int k = 10;

    int temp[k] = {0};

    for (int i = 0; i < n; i++) {
        temp[array[i]]++;
    }
    for (int i = 1; i < k; i++) {
        temp[i] += temp[i - 1];
    }
    for (int i = n - 1; i >= 0; i--) {  
        output[temp[array[i]] - 1] = array[i];
        temp[array[i]]--;  
    }
    for (int i = 0; i < n; i++) {
        std::cout << output[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}
